web/Web/web/CAL/views.py

```diff
@@ -198,3 +199,70 @@ def get_ajax(self, request, *args, **kwargs):
         except CALError as e:
             # TODO: add proper http response for CAL errors
             return JsonResponse({"message": "Ops! CALError."}, status=404)
+
+
+class DSLoggingView(views.CsrfExemptMixin,
+                    views.LoginRequiredMixin,
+                    RetrievalMethodPermissionMixin,
+                    views.JsonRequestResponseMixin,
+                    generic.View):
+    """
+    View to log info of Dynamic Sampling
+    """
+    require_json = False
+
+    def post(self, request, *args, **kwargs):
+        session = self.request.user.current_session.uuid
+
+        try:
+            stratum_num = self.request_json.get(u"stratum_number")
+            stratum_size = self.request_json.get(u"stratum_size")
+            T = self.request_json.get(u"T")
+            N = self.request_json.get(u"N")
+            R = self.request_json.get(u"R")
+            current_sample_size = self.request_json.get(u"n")
+
+            # Get docs in current stratum
```

Comment:
hmm It seems cumbersome to me to make two extra calls to CAL if the stratum already exists.

Reply:
It would be a much nicer solution if  CAL would include the sampled docs in the request itself so that you don't need to make a separate call. 

The complete stratum docs can be a separate function that's is called only when needed.

Reply:
I combine them into one function now.